### ISSUE TITLE
unbound variable not detected

### ISSUE THREAD

**Issue by udf2457** [koalaman/shellcheck#3069]
#### For bugs
version: 0.10.0

#### Here's a snippet or screenshot that shows the problem:

This is a snippet from a longer script....

```sh

#!/usr/bin/env bash

set -uo pipefail

# Get Tailscale nodekeys for Mullvad owned
declare -a mOwnedKeys
for mOwned in "${mullvadOwned[@]}"; do
    nKey=$(${JQ} --arg HOST "${mOwned}" -r '.FilteredPeers[] | select (.Name | startswith($HOST)) | .NodeKey' <<< "${TAILSCALE_MULLVAD}")|| exit 1
    if [[ "${nKey}" != "" ]]; then
        mOwnedKeys+=("${nKey}")
    fi
done

# Sign Mullvad owned nodes
if [[ "${signOwned}" == "y" ]]; then
    if [[ ${#mOwnedKeys[@]} -eq 0 ]]; then
        echo "no Mullvad owned nodes found"
        exit 1
    fi
    for ownedNodekey in "${mOwnedKeys[@]}"; do
        echo "signing Mullvad owned key ${ownedNodekey}"
        ${TAILSCALE} lock sign "${ownedNodekey}"
    done
fi


```

#### Here's what shellcheck currently says:

shellcheck does not highlight any issues

#### Here's what I wanted or expected to see:

When running the script `GNU bash, version 5.2.15(1)-release (aarch64-unknown-linux-gnu)` it fails with an alert ` mOwnedKeys: unbound variable` referencing the line `if [[ ${#mOwnedKeys[@]} -eq 0 ]]; then`

**Comment by brother** [comment]
shellcheck.net does hilight this.

![image](https://github.com/user-attachments/assets/9f50318d-e26e-433f-b3d1-e61b3c1d1a75)
